docs: add catalog entry for CUR (#50)
cds-snc/data-lake: docs/data/catalog/operations/aws/cost-and-usage-report.md

# AWS Cost and Usage Report

Dataset describing how much was spent on AWS services by CDS.

Each row describes the cost of using a particular AWS service (i.e., a line item) within a billing period.

> This dataset is represented in [Superset](https://superset.cdssandbox.xyz/superset/welcome/) as the Physical dataset [`cost_usage_report_by_account`](https://superset.cdssandbox.xyz/explore/?datasource_type=table&datasource_id=68). All of the Virtual datasets in the "Operations / AWS / Cost and Usage" group are derived from it.

Reviewer: Just a minor tweak to the URL:
```suggestion
This dataset is represented in [Superset](https://superset.cdssandbox.xyz/) as the Physical dataset [`cost_usage_report_by_account`](https://superset.cdssandbox.xyz/explore/?datasource_type=table&datasource_id=68). All of the Virtual datasets in the "Operations / AWS / Cost and Usage" group are derived from it.
```

**Keywords:** AWS, Amazon, cost, usage, cost, fees

## Provenance

This dataset is extracted daily from the [Cost and Usage Report 2.0 (CUR 2.0)](https://docs.aws.amazon.com/cur/latest/userguide/table-dictionary-cur2.html) table in [AWS Data Exports](https://docs.aws.amazon.com/cur/latest/userguide/what-is-data-exports.html). Additional information about business units are joined in the pipeline using the `tag_env` field.

More documentation on the pipeline can be found [here](../../../pipelines/operations/aws/cost-and-usage-report.md).

* **Updated:** Daily
* **Steward:** Platform Core Services

> * **Contact:** [Pat Heard](mailto:pat.heard@cds-snc.ca)

Reviewer: Another small adjustment to point to my email (this one is just an alias since people kept using it 😅 )
```suggestion
* **Contact:** [Pat Heard](mailto:patrick.heard@cds-snc.ca)
```

* **Location:** s3://cds-data-lake-transformed-production/operations/aws/cost-usage-report/data/billing_period=YYYY-MM/*.parquet

## Fields

Most field definitions are sourced directly from AWS's [Cost and Usage Report data dictionary](https://docs.aws.amazon.com/cur/latest/userguide/table-dictionary-cur2-reservation.html).

Many columns are grouped together with a common prefix. For example, the `product_comment`, `product_fee_code` and `product_fee_description` fields can all be grouped together as `product` columns.

Example data can be found [here](examples/cost-and-usage-report.csv), as a CSV.

> * **id** (integer) - unique identifier for the line item.
> * **arn** (string) - Amazon Resource Name for resource being billed.
> * **email** (string) - email of user who incurred the cost.
> * **name** (string) - name of user who incurred the cost.
> * **status** (string) - status of the line item, one of "ACTIVE" or "INACTIVE".
> * **joinedmethod** (string) - how the resource was added, one of "CREATED" or "INVITED".
> * **joinedtimestamp** (string) - timestamp of service creation in UTC. Formatted as `YYYY-MM-DD HH:MM:SSz.`.

Reviewer: Just adjusting these since they're joined items that related to the account of each line item.  This is a custom export of data with do from our AWS org account so the documentation for it unfortunately exists all in our heads:

```suggestion
* **id** (integer) - AWS account ID for the line item.
* **arn** (string) - Amazon Resource Name of the AWS account for the resource being billed.
* **email** (string) - email associated with the AWS account for the line item.
* **name** (string) - name of the AWS account for the line item.
* **status** (string) - status of the line item's AWS account, one of "ACTIVE" or "INACTIVE".
* **joinedmethod** (string) - how the AWS account was was added to the AWS organization, one of "CREATED" or "INVITED".
* **joinedtimestamp** (string) - timestamp of AWS account's creation in UTC. Formatted as `YYYY-MM-DD HH:MM:SSz.`.
```

Definitely open to adjustments or tweaks on these!
